ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -152,6 +153,27 @@
         (get-unique-native-id context subscription))
       native-id)))
 
+(defn check-subscriber-id
+  "If subscriber id is provided, checks that token-user has appropriate ACLs.
+  If subscriber id is not provided, then the token-user themself is used"
+  [request-context concept provider-id token-user]
+  (let [parsed-metadata (json/parse-string (:metadata concept) true)]
```

Comment:
This can probably be simplified with an `if-let` to combine the checks. We try to avoid nested `let` when possible.

Reply:
Replacement functions do not use any nested let/if logic for readability